### DISCUSSION TITLE
Overwrite Error

### DISCUSSION THREAD

**Discussion by kspitzer14** [grahampugh/erase-install#118] [Q&A]
I am getting an error on some of my test machines. It looks like it is failing the installer version check but I can't seem to understand why. Using the script to do upgrades from Catalina to Big Sur so I know that the installer build is higher than the system build. I am using the overwrite flag, but the error seems to suggest it isn't being seen. Anyone have an idea how to resolve.

Script result: 
   [erase-install] v0.20.1 script execution started: Thu Aug 19 16:54:14 PDT 2021
   [erase-install] Caffeinating this script (pid=17466)
45000000
   [free_space_check] OK - 1895312484 KB free disk space detected
   [erase-install] Looking for existing installer
   [find_existing_installer] Installer sparse image found at /Library/Management/erase-install/Install_macOS_11.5.2-20G95.sparseimage.
/dev/disk2          	GUID_partition_scheme          	
/dev/disk2s1        	EFI                            	
/dev/disk2s2        	Apple_HFS                      	/Volumes/Install_macOS_11.5.2-20G95
   [check_installer_is_valid] Checking validity of .
   [check_installer_is_valid] Using DTSDKBuild value from Info.plist
2021-08-19 16:54:14.957 defaults[17507:2559444] 
The domain/default pair of (/Contents/Info.plist, DTSDKBuild) does not exist
   [check_installer_is_valid] Installer:  ; System: 19H1323 : invalid build.
   [erase-install] ERROR: Invalid installer is present. Run with --overwrite option to ensure that a valid installer is obtained.

**Comment by grahampugh** [comment]
I suspect you have an incomplete download. Can you show me the command you used?

**Comment by kspitzer14** [comment]
Here is the script copied from our jamf server, the parameters set in jamf, and the log file from one of the machines that it is failing on. We have tested on ~70 machines and had about 5 that have had similar results.
[failure.zip](https://github.com/grahampugh/erase-install/files/7034102/failure.zip)

**Reply by mtucker299** [reply]
I am seeing something similar on Montery to Ventura. 

Script: 
 /Library/Management/erase-install/erase-install.sh --reinstall --os=13 --update  --current-user --min-drive-space=60 --check-power --depnotify --cleanup-after-use

Result log: 

[erase-install] v27.1 script execution started: Fri Dec  9 16:37:52 CST 2022
   [erase-install] Caffeinating this script (pid=2940)
   [check_free_space] OK - 447 GB free/purgeable disk space detected
   [check_power_status] OK - AC power detected
   [erase-install] Looking for existing installer app or pkg
   [find_existing_installer] Installer sparse image found at /Library/Management/erase-install/Install_macOS_13.0.1-22A400.sparseimage.
   [check_installer_is_valid] Checking validity of .
   [check_installer_is_valid] Using DTSDKBuild value from Info.plist
   [check_installer_is_valid] Installer Info.plist could not be found!
   [check_installer_is_valid] Build of existing installer could not be found, so it is assumed to be invalid.
   [erase-install] ERROR: Invalid installer is present. Run with --overwrite option to ensure that a valid installer is obtained.

   [erase-install] attempting to terminate the 'caffeinate' process - Termination message indicates success
/Library/Management/erase-install/erase-install.sh: line 1046:  2960 Terminated: 15          /usr/bin/caffeinate -dimsu -w $$

   [finish] Script exit code: 1

**Reply by mtucker299** [reply]
> Here is the script copied from our jamf server, the parameters set in jamf, and the log file from one of the machines that it is failing on. We have tested on ~70 machines and had about 5 that have had similar results. [failure.zip](https://github.com/grahampugh/erase-install/files/7034102/failure.zip)

I just looked in the install location of erase-install (in my case /Library/Management/erase-install/) and saw the offending "Install_macOS_13.0.1-22A400.sparseimage" file which I deleted. 

Now it runs. 

Might see if there is a file there in yours.
